src/modules/file/file.repository.ts

```diff
@@ -893,4 +898,50 @@ export class SequelizeFileRepository implements FileRepository {
 
     return Number(result[0]['total']) as unknown as number;
   }
+
+  async sumFileSizesChangesBetweenDates(
+    userId: FileAttributes['userId'],
+    sinceDate: Date,
+    untilDate?: Date,
+  ): Promise<number> {
+    const timeCondition = {
+      [Op.gte]: sinceDate,
+      ...(untilDate ? { [Op.lte]: untilDate } : null),
+    };
+
+    const result = await this.fileModel.findAll({
+      attributes: [
+        [
+          Sequelize.literal(`
+            SUM(
+              CASE 
+                WHEN status = 'DELETED' AND date_trunc('day', created_at) = date_trunc('day', updated_at) THEN 0
```

Comment:
If we choose from 3 days ago to now, the record will appear twice? (one for its creation and one for the removal?) Otherwise a file created yesterday and removed today will count as a negative delta, when it should be 0. Am I missing something?